tools/chocolateyinstall.ps1

```diff
@@ -1,40 +1,44 @@
-Install-ChocolateyZipPackage -PackageName $packageName -Url $url -Url64bit $url64 -UnzipLocation $toolsDir -Checksum $checkSum -ChecksumType $checkSumType -Checksum64 $checkSum -ChecksumType64 $checkSumType
-
-if ([string]::IsNullOrEmpty($pp['Token'])) {
-  Write-Output "RG Supervision agent installation cannot be processed without <Token> mandatory parameter."
-  Write-Output "You should have used the following command:"
-  Write-Output "choco install rgsupervision --params "'/Token:value(string) /Node:value(int)'""
-  Write-Output "If you do not have your deployment token yet, check your Deployment section within your user profile management page."
-  Write-Output "You will then be able to finish the installation by running the following command:"
-  Write-Output "$($exeFilePath) --action register --login token@token.tk --password <YourToken> --node #$($pp['Node'])"
+
+if (!$pp["Token"]) { $pp["Token"] = Read-Host "Please enter your deployment token:" }
+if (!$pp["Node"]) { $pp["Node"] = Read-Host "Please enter the node ID where agent should be deployed (default: root):" }
+if (!$pp["Node"]) { $pp["Node"] = "root" }
+
+##### Attempting either binary install or (if no Token provided) simple zip extract
+
+if ([string]::IsNullOrEmpty($pp["Token"])) {
+  Write-Error "RG Supervision agent installation cannot be processed without <Token> mandatory parameter."
+  Write-Error "You should have used the following command:"
+  Write-Error "choco install rgsupervision --params `"'/Token:value(string) /Node:value(int)'`""
+  Write-Error "If you do not have your deployment token yet, check your Deployment section within your user profile management page."
 } else {
   Write-Output "Starting agent install using deployment token: $($pp['Token'])"
+  Write-Output $(if ($pp["ExpectedHostName"]) {"Deploying (custom hostname) on node #$($pp['Node'])"} else {"Deploying on node #$($pp['Node'])"})
 
-  if ($pp['ExpectedHostName']) {
-    Write-Output "Deploying (custom hostname) on node #$($pp['Node'])"
-    & $exeFilePath --action register --login "token@token.tk" --password "$($pp['Token'])" --node "#$($pp['Node'])" --expected-host-name "$($pp['ExpectedHostName'])"
-  } else {
-    Write-Output "Deploying on node #$($pp['Node'])"
-    & $exeFilePath --action register --login "token@token.tk" --password "$($pp['Token'])" --node "#$($pp['Node'])"
+  $silentArgs = "--action register --login token@token.tk --password $($pp['Token']) --node #$($pp['Node'])"
+  if ($pp["ExpectedHostName"]) { $silentArgs += " --expected-host-name $($pp['ExpectedHostName'])" } # ExpectedHostName parameter mainly used for internal debug or on premise requirements
+
+  $packageArgs = @{
+    packageName    = $packageName
+    fileType       = "exe"
+    file           = $fileLocation
+    file64         = $fileLocation
+    checksum       = $checkSum
+    checksum64     = $checkSum
+    checksumType   = $checkSumType
+    silentArgs     = $silentArgs
+    validExitCodes = @(0, 3010, 1641)
```

Comment:
Same here?

Reply:
Hmmm probably kept it from chocolatey doc but after quick check it looks like MSI packages error codes
Don't know RG-Setup.exe possible exit codes, should I keep 0 and 1 ?

Reply:
![image](https://user-images.githubusercontent.com/18307823/185346380-63412de5-8dfe-4b13-a8ea-cdc178bb0c49.png)

Reply:
validExitCode should cover "valid exit code for success", or any return status?